pepper-apis/src/main/java/org/broadinstitute/ddp/customexport/db/dao/CustomExportDao.java

```diff
@@ -0,0 +1,93 @@
+package org.broadinstitute.ddp.customexport.db.dao;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.Set;
+
+import org.broadinstitute.ddp.customexport.db.dto.CustomExportDto;
+import org.broadinstitute.ddp.db.DBUtils;
+import org.broadinstitute.ddp.db.DaoException;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.config.RegisterConstructorMapper;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+public interface CustomExportDao extends SqlObject {
+
+    default Instant getLastCompleted(long studyId) {
+        Optional<Timestamp> optionalLastCompleted = findExportLastCompletedByStudyId(studyId);
+        if (optionalLastCompleted.isPresent()) {
+            return optionalLastCompleted.get().toInstant();
+        } else {
+            // Return 0 but make sure to add this to the database
+            insertExport(studyId, Instant.EPOCH);
+            findExportDtoByStudyId(studyId).orElseThrow(() ->
+                    new DaoException("Could not find newly created custom export with id " + studyId));
+            return Instant.EPOCH;
+        }
+    }
+
+    default void updateLastCompleted(Instant lastCompleted, long studyId) {
+        int lastCompletedByStudyId = updateLastCompletedByStudyId(lastCompleted, studyId);
+        DBUtils.checkUpdate(1, (int)lastCompletedByStudyId);
+    }
+
+    @SqlUpdate("insert into custom_export (study_id, custom_export_last_completed) values (:studyId, :lastCompleted)")
+    void insertExport(
+            @Bind("studyId") long studyId,
+            @Bind("lastCompleted") Instant lastCompleted
+    );
+
+    @SqlQuery("select * from custom_export where study_id = :id")
+    @RegisterConstructorMapper(CustomExportDto.class)
+    Optional<CustomExportDto> findExportDtoByStudyId(@Bind("id") long id);
+
+    @SqlQuery("select custom_export_last_completed from custom_export where study_id = :id")
+    Optional<Timestamp> findExportLastCompletedByStudyId(@Bind("id") long id);
+
+    @SqlUpdate("update custom_export set custom_export_last_completed = :lastCompleted where study_id = :studyId")
+    int updateLastCompletedByStudyId(@Bind("lastCompleted") Instant lastCompleted, @Bind("studyId") long studyId);
+
+    @SqlQuery("SELECT CASE WHEN ( "
+            + "SELECT COUNT(study_activity_id) FROM ( "
+            + "SELECT sa.study_activity_id FROM study_activity AS sa  "
```

Comment:
Shouldn't this query look for _activity instance ids_ instead of `study_activity_id`? `study_activity` are the activity definitions, which doesn't change, whereas `activity_instance` holds instances for each user. Also beware that each activity instance will have multiple `activity_instance_status` rows (especially if they're completed), so we're going to end up with a cross-product here, but I guess that's okay since we're just trying to figure out if there's at least one new instance.